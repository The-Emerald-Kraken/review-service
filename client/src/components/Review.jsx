/* eslint-disable no-underscore-dangle */
/* eslint-disable import/extensions */
import React from 'react';
import styled from 'styled-components';
import moment from 'moment';
import PropTypes from 'prop-types';
import Star from './Star.jsx';
import Fit from './Fit.jsx';
import Helpful from './Helpful.jsx';

const Review = ({ review, patchData }) => (
  <FlexContainer>
    <FlexBoxL>
      <p><b>{review.nickname}</b></p>
      <p>{review.location}</p>
    </FlexBoxL>
    <FlexBoxC>
      <Wrapper>
        <Star rating={review.rating} />
        {moment(review.createdAt).fromNow()}
      </Wrapper>

      <Title>{review.title}</Title>

      <p>{review.body}</p>
      <p>
        <b>Height </b>
        {review.height}
      </p>
      <p>
        <b>Weight </b>
        {review.weight}
      </p>
      <p>
        <b>Age </b>
        {review.age}
      </p>
      <p>
        <b>Primary Usage </b>
        <Wrapper>
          {review.tags.map((tag) => (`${tag} `))}
        </Wrapper>
      </p>
      {review.images.map((image) => <JessesProblem key={image.id} inputSrc={image.url} />)}
      <Helpful
        helpful={review.helpful}
        idd={review._id}
        patchData={patchData}
      />
    </FlexBoxC>
    <FlexBoxR>
      <Title2>Overall Fit Rating</Title2>
      <Fit rating={review.fit} />
    </FlexBoxR>

  </FlexContainer>
);

Review.propTypes = {
  review: PropTypes.shape().isRequired,
  patchData: PropTypes.func.isRequired,
};

const FlexContainer = styled.section`
display: flex;
font-family: Stuart,Georgia,serif;
font-size: 16px;
border-bottom: groove;
`;
const FlexBoxL = styled.div`
  margin: 20px;
  margin-top: 0;
  padding: 20px;

`;
const FlexBoxC = styled.div`
  margin: 20px;
  margin-right: 70px;
  margin-left: 70px;
  margin-top: 0;
  padding: 20px;

`;
const FlexBoxR = styled.div`
padding-top: 300px;
margin: 10px;
padding: 20px;
padding-top: 90px;
`;
const JessesProblem = styled.img`
  width: 150px;
  height: 150px;
  background-size: auto;
  background-repeat: no-repeat;
  background-image: url(${(props) => props.inputSrc});
  position: relative;
  margin-right: 1.5px;
  margin -left: 1.5px;
`;
const Title = styled.h2`
    color: #292929;
    font-size: 19px;
    line-height: 20px;
    font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
    font-style: normal;
    font-weight: 700;
    `;

const Title2 = styled.h2`
  display: block!important;
  margin-bottom: .5em!important;
  font-size: 15px;
  line-height: 20px;
  font-style: normal;
  font-weight: 400;
`;
const Wrapper = styled.span`
  display:inline-block;
`;

export default Review;
