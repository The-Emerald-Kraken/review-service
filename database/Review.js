/* eslint-disable no-unused-vars */
const mongoose = require('mongoose');
const db = require('./index.js');
// mongoose.Promise = global.Promise;

const reviewsSchema = new mongoose.Schema({
  product_id: Number,
  review_id: Number,
  nickname: String,
  rating: Number,
  title: String,
  body: String,
  images: [{
    id: Number,
    url: String,
  }],
  videos: [{
    id: Number,
    url: String,
  }],
  tags: [String],
  height: String,
  weight: String,
  fit: String,
  age: String,
  location: String,
  email: String,
  helpful: {
    yes: Number,
    no: Number,
  },
},
{
  timestamps: true,
});

const Review = mongoose.model('reviews', reviewsSchema);

module.exports = Review;
