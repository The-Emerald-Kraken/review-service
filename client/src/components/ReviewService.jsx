/* eslint-disable no-underscore-dangle */
/* eslint-disable import/extensions */
import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import axios from 'axios';
import PropTypes from 'prop-types';
import Review from './Review.jsx';
import LoadMore from './LoadMore.jsx';
import ReviewStatsContainer from './ReviewStatsContainer.jsx';

function ReviewService({ start }) {
  const [display, setDisplay] = useState(5);
  const [reviews, setReviews] = useState([]);
  const [avg, setAvg] = useState({ rating: 0, fit: 0 });
  const [currentItem] = useState(start);
  const [show] = useState(true);

  // did comopoent mount replacement

  const FetchData = (requests, id) => {
    axios.get(`/api/products/reviews/${id}/${requests}`)
      .then(({ data }) => setReviews(data))
      .catch((err) => (err));
  };
  const FetchAvgData = (request) => {
    axios.get(`/api/products/reviews/avg/${request}`)
      .then(({ data }) => setAvg(data))
      .catch((err) => (err));
  };

  const PatchData = (selected, id) => {
    axios.patch(`/api/products/reviews/${selected}/${id}`)
      .then(() => FetchData(display, currentItem))
      .catch((err) => err);
  };

  useEffect(() => {
    FetchAvgData(currentItem);
  }, [currentItem]);

  useEffect(() => {
    FetchData(display, currentItem);
  }, [display]);

  // did compoenet update replacement
  useEffect(() => {
    if (show) {
      FetchData(display, currentItem);
    }
  }, [display || currentItem]);
  return (
    <Wrapper>
      <ReviewStatsContainer avgReview={avg} />
      <ul>
        {reviews.map((review) => (
          <Review
            key={review._id}
            review={review}
            patchData={PatchData}
          />
        ))}
      </ul>
      {show ? (
        <LoadMore
          changeDisplay={setDisplay}
          currentDisplay={display}
        />
      ) : null}
    </Wrapper>
  );
}

ReviewService.propTypes = {
  start: PropTypes.number.isRequired,
};

const Wrapper = styled.section`
padding: 4em;
background: whitesmoke;
width: 50%;
margin: auto
`;

export default ReviewService;
