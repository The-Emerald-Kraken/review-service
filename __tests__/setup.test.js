import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import ReviewService from '../client/src/components/ReviewService';
import Review from '../client/src/components/Review';
import Helpful from '../client/src/components/Helpful';
import Fit from '../client/src/components/Fit';
import LoadMore from '../client/src/components/LoadMore';
import Star from '../client/src/components/Star';
import sampleReview from '../sampleData.js'

describe('A suite example using Snapshot',  () => {
  it('renders Reviews Services correctly', () => {
    const tree = renderer
      .create(<ReviewService start={1}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Fit Display Component', () => {
    const tree = renderer
      .create(<Fit rating={sampleReview.fit} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Helpful Button Component', () => {
    const tree = renderer
      .create(<Helpful
        helpful={sampleReview.helpful}
        idd={sampleReview._id}
        patchData={()=>(null)}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Load More Button Component', () => {
    const tree = renderer
      .create(<LoadMore
        changeDisplay={()=>(null)}
        currentDisplay={1}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Review Display Component', () => {
    const tree = renderer
      .create(<Review
        review={sampleReview}
        patchData={()=>(null)}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Star Review Component', () => {
    const tree = renderer
      .create(<Star rating={sampleReview.rating} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


});

// to my understanding, shallow make it so I do not have to 'mount' it to a real page
describe('A test suite example using enzyme', function() {
  xit('should render ReviewService without throwing an error', function() {
    expect(shallow(<ReviewService start={1} />).contains('Reviews')).toBe(true);
  });

});