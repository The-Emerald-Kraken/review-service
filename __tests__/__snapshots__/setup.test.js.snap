// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite example using Snapshot renders Reviews Services correctly 1`] = `
<section
  className="sc-cxNIbT kBJFUG"
>
  <div
    className="sc-fKgIGh jmiSIv"
  >
    <h1
      className="sc-bCwgka bZJqHK"
    >
      Reviews
    </h1>
    <button
      className="sc-iwaifL kByyyd"
    >
      Write a review
    </button>
    <section
      className="sc-fFSRdu eEQBRj"
    >
      <div
        className="sc-kfYqjs fgrQAH"
      >
        <table
          className="sc-bkbjAj favYKL"
        >
          <thead>
            <tr>
              <td>
                Rating Snapshot
              </td>
            </tr>
            <tr>
              <td>
                Select a row below to filter reviews.
              </td>
            </tr>
            <tr>
              <td
                className="sc-hHEjAm jJbWbE"
              >
                5 ★
                <div
                  className="sc-hBMVcZ gGMrHL"
                  onClick={[Function]}
                >
                  <div
                    className="sc-fnVYJo deJmWL"
                  />
                </div>
              </td>
            </tr>
            <tr>
              <td
                className="sc-hHEjAm jJbWbE"
              >
                4 ★
                <div
                  className="sc-hBMVcZ gGMrHL"
                  onClick={[Function]}
                >
                  <div
                    className="sc-fnVYJo deJmWL"
                  />
                </div>
              </td>
            </tr>
            <tr>
              <td
                className="sc-hHEjAm jJbWbE"
              >
                3 ★
                <div
                  className="sc-hBMVcZ gGMrHL"
                  onClick={[Function]}
                >
                  <div
                    className="sc-fnVYJo deJmWL"
                  />
                </div>
              </td>
            </tr>
            <tr>
              <td
                className="sc-hHEjAm jJbWbE"
              >
                2 ★
                <div
                  className="sc-hBMVcZ gGMrHL"
                  onClick={[Function]}
                >
                  <div
                    className="sc-fnVYJo deJmWL"
                  />
                </div>
              </td>
            </tr>
            <tr>
              <td
                className="sc-hHEjAm jJbWbE"
              >
                1 ★
                <div
                  className="sc-hBMVcZ gGMrHL"
                  onClick={[Function]}
                >
                  <div
                    className="sc-fnVYJo deJmWL"
                  />
                </div>
              </td>
            </tr>
          </thead>
        </table>
      </div>
      <div
        className="sc-dlMBXb fkaCyn"
      >
        <table
          className="sc-bkbjAj favYKL"
        >
          <caption>
            Average Customer Ratings
          </caption>
          <thead>
            <tr>
              <td>
                Overall
              </td>
              <td
                className="sc-iemXMA kltcnJ"
              >
                <div
                  className="sc-bdnylx eTIvCG"
                />
              </td>
              <td
                className="sc-iemXMA kltcnJ"
              >
                0
              </td>
            </tr>
            <tr>
              <td
                className="sc-dIvqjp bAuUCd"
              >
                Overall Fit Rating
              </td>
              <td
                className="sc-iemXMA kltcnJ"
              >
                <div
                  className="sc-jSFkmK lnDnEi"
                >
                  <div
                    className="sc-gtssRu dWODPw"
                  >
                    <div
                      className="sc-dlnjPT eUUHfX"
                    />
                    <span
                      className="sc-gKAblj dJFrAv"
                    />
                  </div>
                  <span
                    className="sc-hKFyIo fNFVfG"
                  >
                    Runs Small
                  </span>
                  <span
                    className="sc-eCApGN dNfuhr"
                  >
                    Runs Large
                  </span>
                </div>
              </td>
            </tr>
          </thead>
        </table>
      </div>
    </section>
  </div>
  <ul />
  <button
    className="sc-ksluoS djYkNW"
    onClick={[Function]}
  >
    Load More
  </button>
</section>
`;

exports[`A suite example using Snapshot renders correctly Fit Display Component 1`] = `
<div
  className="sc-jSFkmK lnDnEi"
>
  <div
    className="sc-gtssRu dWODPw"
  >
    <div
      className="sc-dlnjPT dUUXmW"
    />
    <span
      className="sc-gKAblj dJFrAv"
    />
  </div>
  <span
    className="sc-hKFyIo fNFVfG"
  >
    Runs Small
  </span>
  <span
    className="sc-eCApGN dNfuhr"
  >
    Runs Large
  </span>
</div>
`;

exports[`A suite example using Snapshot renders correctly Helpful Button Component 1`] = `
<span>
  Helpful?
  <button
    className="sc-iCoHVE hDOOVg"
    onClick={[Function]}
  >
    Yes · 
    1
  </button>
  <button
    className="sc-iCoHVE hDOOVg"
    onClick={[Function]}
  >
    No · 
    2
  </button>
  <button
    className="sc-iCoHVE hDOOVg"
  >
    Report as inappropriate
  </button>
</span>
`;

exports[`A suite example using Snapshot renders correctly Load More Button Component 1`] = `
<button
  className="sc-ksluoS djYkNW"
  onClick={[Function]}
>
  Load More
</button>
`;

exports[`A suite example using Snapshot renders correctly Review Display Component 1`] = `
<section
  className="sc-fujyUd cfrUuJ"
>
  <div
    className="sc-pNWxx glRlCA"
  >
    <p>
      <b>
        theGreekest
      </b>
    </p>
    <p>
      New York
    </p>
  </div>
  <div
    className="sc-jrsJCI eQGrVN"
  >
    <span
      className="sc-bqGHjH gBkTZR"
    >
      <div
        className="sc-bdnylx ddxNlS"
      />
      3 years ago
    </span>
    <h2
      className="sc-crzoUp ixgpyj"
    >
      The Greeks Strike Again
    </h2>
    <p>
      GreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreek
    </p>
    <p>
      <b>
        Height 
      </b>
      Shorter than 5'
    </p>
    <p>
      <b>
        Weight 
      </b>
      375 - 400 lbs
    </p>
    <p>
      <b>
        Age 
      </b>
      18 to 24
    </p>
    <p>
      <b>
        Primary Usage 
      </b>
      <span
        className="sc-bqGHjH gBkTZR"
      >
        Shirt 
        Pratical 
        #streetWear 
      </span>
    </p>
    <img
      className="sc-iqAbSa jTOThj"
    />
    <span>
      Helpful?
      <button
        className="sc-iCoHVE hDOOVg"
        onClick={[Function]}
      >
        Yes · 
        1
      </button>
      <button
        className="sc-iCoHVE hDOOVg"
        onClick={[Function]}
      >
        No · 
        2
      </button>
      <button
        className="sc-iCoHVE hDOOVg"
      >
        Report as inappropriate
      </button>
    </span>
  </div>
  <div
    className="sc-kEqYlL hWtfSf"
  >
    <h2
      className="sc-dIsAE gEkXsQ"
    >
      Overall Fit Rating
    </h2>
    <div
      className="sc-jSFkmK lnDnEi"
    >
      <div
        className="sc-gtssRu dWODPw"
      >
        <div
          className="sc-dlnjPT dUUXmW"
        />
        <span
          className="sc-gKAblj dJFrAv"
        />
      </div>
      <span
        className="sc-hKFyIo fNFVfG"
      >
        Runs Small
      </span>
      <span
        className="sc-eCApGN dNfuhr"
      >
        Runs Large
      </span>
    </div>
  </div>
</section>
`;

exports[`A suite example using Snapshot renders correctly Star Review Component 1`] = `
<div
  className="sc-bdnylx ddxNlS"
/>
`;
